### PR TITLE
Ignore containers not managed by cilium

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

**Review by aanm — CHANGES_REQUESTED** [review]
Needs a mutex for the map.

**Comment by tgraf** [comment]
How are you ensuring concurrency in `updateContainer` at all? This needs more than just a mutex around the map.

**Comment by aanm** [comment]
All stuff inside `updateContainer` are "concurrency-free". For example `d.setEndpointSecLabel` has a lock inside of it.

**Comment by tgraf** [comment]
That's not enough. The same ID can be handled by multiple go routines which then call `setEndpointSecLabel()` on the same ID. Although the lock avoids parallelism of `setEndpointSecLabel()` itself, it does not avoid multiple calls to `setEndpointSecLabel()` which then leaves stale entries in the endpoints table.

File: daemon/daemon/daemon.go [hunk 1]
```diff
@@ -499,3 +502,27 @@ func (d *Daemon) Update(opts option.OptionMap) error {
...
+	d.ignoredMutex.RLock()
+	_, ok := d.ignoredContainers[id]
+	d.ignoredMutex.RUnlock()
+
+	return ok
+}
+
+func (d *Daemon) StartIgnoringContainer(id string) {
+	d.ignoredMutex.Lock()
+	if _, ok := d.ignoredContainers[id]; ok {
```
Comment (aanm) [inline daemon/daemon/daemon.go]: Since the value is not a pointer you can directly do.
```go
d.ignoredContainers[id]++
```

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by aanm — APPROVED** [review]

**Pull request merged**